mslib/mswms/mpl_1sec_styles.py

```diff
@@ -0,0 +1,357 @@
+import numpy as np
+
+from mslib.mswms.mpl_1sec import Abstract1DSectionStyle
+import mslib.thermolib as thermolib
+from mslib.utils import convert_to
+
+
+class OS_TemperatureStyle_01(Abstract1DSectionStyle):
+    """
+    1D section of temperature.
+    """
+
+    name = "OS_T01"
+    title = "Temperature (K) 1D Section"
+    abstract = "Temperature (K)"
+
+    # Variables with the highest number of dimensions first (otherwise
+    # MFDatasetCommonDims will throw an exception)!
+    required_datafields = [
+        ("ml", "air_pressure", "Pa"),
+        ("ml", "air_temperature", "K")]
+
+    def _plot_style(self, color):
+        """
+        Make a temperature 1D section.
+        """
+        ax = self.ax
+        self.y_values = self.data["air_temperature"]
+
+        numpoints = len(self.lats)
+        ax.plot(range(numpoints), self.y_values, color.replace("0x", "#"))
+        self._latlon_setup()
+
+
+class OS_RelativeHumdityStyle_01(Abstract1DSectionStyle):
+    """
+    1D section of relative humidity.
+    """
+
+    name = "OS_RH01"
+    title = "Relative Humdity (%) 1D Section"
+    abstract = "Relative humdity (%)"
+
+    # Variables with the highest number of dimensions first (otherwise
+    # MFDatasetCommonDims will throw an exception)!
+    required_datafields = [
+        ("ml", "air_pressure", "Pa"),
+        ("ml", "air_temperature", "K"),
+        ("ml", "specific_humidity", "kg/kg")]
+
+    def _prepare_datafields(self):
+        """
+        Computes relative humdity.
+        """
+        self.data["relative_humidity"] = thermolib.rel_hum(
+            self.data['air_pressure'], self.data["air_temperature"],
+            self.data["specific_humidity"])
+
+    def _plot_style(self, color):
+        ax = self.ax
+        self.y_values = self.data["specific_humidity"]
```

Comment:
See, all classes run down to plotting one data field. No need for so many classes...? I am not sure about this, so. You have more insight into the implementation, but for this type we need either a set of generic classes or something like the make_generic function of the other styles.